hera_cal/datacontainer.py

```diff
@@ -84,12 +92,80 @@ def __delitem__(self, key):
         else:
             raise ValueError('only supports setting (ant1, ant2, pol) keys')
 
+    def concat(self, D, axis=0):
+        """ concatenates DataContainers across an axis """
+        # check type of D
+        if isinstance(D, DataContainer):
+            # turn into list if not already a list
+            D = [D]
+        if axis == 0:
+            # check 1 axis is identical for all D
+            for d in D:
+                if d[d.keys()[0]].shape[1] != self.__getitem__(self.keys()[0]).shape[1]:
+                    raise ValueError("[1] axis of dictionary values aren't identical in length")
+
+        if axis == 1:
+            # check 0 axis is identical for all D
+            for d in D:
+                if d[d.keys()[0]].shape[0] != self.__getitem__(self.keys()[0]).shape[0]:
+                    raise ValueError("[0] axis of dictionary values aren't identical in length")
+
+        # start new object
+        newD = odict()
+
+        # get shared keys
+        keys = set()
+        for d in D: keys.update(d.keys())
+
+        # iterate over D keys
+        for i, k in enumerate(keys):
+            if self.__contains__(k):
+                if axis == 0:
+                    newD[k] = np.concatenate([self.__getitem__(k)] + map(lambda d: d[k], D), axis=0)
+                elif axis == 1:
+                    newD[k] = np.concatenate([self.__getitem__(k)] + map(lambda d: d[k], D), axis=1)
+
+        return DataContainer(newD)
   
-  
+    def __add__(self, D):
+        """ adds values of two DataContainers together """
+        # check time and frequency structure matches
+        if D[D.keys()[0]].shape[0] != self.__getitem__(self.keys()[0]).shape[0]:
```

Comment:
This shape verification code is used a lot. Perhaps you want to break it out in its own function?

Reply:
meh